### PR TITLE
Added some initial support for free toolchains

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

**Comment by SergiusTheBest** [comment]
That's interesting! Do you build drivers with mingw? Could you provide some info about your use case?

**Comment by sgeto** [comment]
I like the idea.
I would personally put the MinGW/ReactOS stuff in a separate file and call `include()`, or make them a separate find module altogether

**Comment by SergiusTheBest** [comment]
@sfhacker Very nice, please keep us posted!

**Comment by DiskGetor** [comment]
Mingw, sys, sure? If the google go language can build sys file,that's best for me.

**Comment by sfhacker** [comment]
@DiskGetor Can you rephrase your last post in plain English?
